packages/indexer-common/src/allocations/__tests__/tap.test.ts

```diff
@@ -125,25 +151,209 @@ describe('TAP', () => {
     timeout,
   )
 
+  test('should revert the rav request', async () => {
+    // we have a redeemed non-final rav in our database
+    const nowSecs = Math.floor(Date.now() / 1000)
+    // redeemed rav but non-final
+    const ravList = [
+      createLastNonFinalRav(
+        ALLOCATION_ID_3,
+        SENDER_ADDRESS_1,
+        new Date((nowSecs - 1) * 1000),
+      ),
+      createLastNonFinalRav(ALLOCATION_ID_3, SENDER_ADDRESS_2, new Date(nowSecs * 1000)),
+      createLastNonFinalRav(
+        ALLOCATION_ID_3,
+        SENDER_ADDRESS_3,
+        new Date((nowSecs + 1) * 1000),
+      ),
+    ]
+
+    await queryFeeModels.receiptAggregateVouchers.bulkCreate(ravList)
+
+    // it's not showing on the subgraph on a specific point in time
+    // the timestamp of the subgraph is greater than the receipt id
+    // should revert the rav
+    await receiptCollector['revertRavsRedeemed'](ravList, nowSecs - 1)
+
+    let lastRedeemedRavs = await queryFeeModels.receiptAggregateVouchers.findAll({
+      where: {
+        last: true,
+        final: false,
+        redeemedAt: {
+          [Op.ne]: null,
+        },
+      },
+    })
+    expect(lastRedeemedRavs).toEqual([
+      expect.objectContaining(ravList[0]),
+      expect.objectContaining(ravList[1]),
+      expect.objectContaining(ravList[2]),
+    ])
+
+    await receiptCollector['revertRavsRedeemed'](ravList, nowSecs)
+
+    lastRedeemedRavs = await queryFeeModels.receiptAggregateVouchers.findAll({
+      where: {
+        last: true,
+        final: false,
+        redeemedAt: {
+          [Op.ne]: null,
+        },
+      },
+    })
+    expect(lastRedeemedRavs).toEqual([
+      expect.objectContaining(ravList[1]),
+      expect.objectContaining(ravList[2]),
+    ])
+
+    await receiptCollector['revertRavsRedeemed'](ravList, nowSecs + 1)
+
+    lastRedeemedRavs = await queryFeeModels.receiptAggregateVouchers.findAll({
+      where: {
+        last: true,
+        final: false,
+        redeemedAt: {
+          [Op.ne]: null,
+        },
+      },
+    })
+    expect(lastRedeemedRavs).toEqual([expect.objectContaining(ravList[2])])
+
+    await receiptCollector['revertRavsRedeemed'](ravList, nowSecs + 2)
+
+    lastRedeemedRavs = await queryFeeModels.receiptAggregateVouchers.findAll({
+      where: {
+        last: true,
+        final: false,
+        redeemedAt: {
+          [Op.ne]: null,
+        },
+      },
+    })
+    expect(lastRedeemedRavs).toEqual([])
+  })
+
+  test('should not revert the rav request, allocation_id not in the list ', async () => {
+    // we have a redeemed non-final rav in our database
```

Comment:
Is this comment to be removed?

Reply:
I usually create english sentenses to structure the test, and then implement the code. But this felt disconnected, gonna remove it.